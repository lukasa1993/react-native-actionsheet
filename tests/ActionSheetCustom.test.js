import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import ExampleA from '../Examples/Basic/ExampleA'
import ExampleB from '../Examples/Basic/ExampleB'

test('ExampleA render correctly', () => {
  const tree = renderer.create(
    <ExampleA />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('ExampleB render correctly', () => {
  const tree = renderer.create(
    <ExampleB />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
