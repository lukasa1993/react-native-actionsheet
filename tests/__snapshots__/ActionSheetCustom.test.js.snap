// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleA render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    I like 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#38f",
        "color": "#fff",
        "marginBottom": 10,
        "paddingBottom": 15,
        "paddingTop": 15,
        "textAlign": "center",
        "width": 200,
      }
    }
  >
    Example A
  </Text>
</View>
`;

exports[`ExampleB render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "marginBottom": 20,
      }
    }
  >
    I like 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Object {
        "backgroundColor": "#38f",
        "color": "#fff",
        "marginBottom": 10,
        "paddingBottom": 15,
        "paddingTop": 15,
        "textAlign": "center",
        "width": 200,
      }
    }
  >
    Example B
  </Text>
  <Modal
    animationType="none"
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#000",
            "bottom": 0,
            "left": 0,
            "opacity": 0.4,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
      <View
        style={
          Object {
            "alignSelf": "flex-end",
            "backgroundColor": "#e5e5e5",
            "flex": 1,
            "height": 338.5,
            "transform": Array [
              Object {
                "translateY": 338.5,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "height": 40,
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontSize": 18,
              }
            }
          >
            Which one do you like?
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "height": 40,
              "justifyContent": "center",
              "paddingBottom": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#8f8f8f",
                "fontSize": 12,
              }
            }
          >
            custom message custom message custom message custom message custom message custom message 
          </Text>
        </View>
        <ScrollView
          contentContainerStyle={undefined}
          scrollEnabled={false}
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 0.5,
                  },
                  Object {},
                ],
              ]
            }
            testID={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 18,
                  },
                  Object {
                    "color": "#007aff",
                  },
                ]
              }
            >
              Apple
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 0.5,
                  },
                  Object {},
                ],
              ]
            }
            testID={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "yellow",
                }
              }
            >
              Banana
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 0.5,
                  },
                  Object {},
                ],
              ]
            }
            testID={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "fontSize": 18,
                  },
                  Object {
                    "color": "#007aff",
                  },
                ]
              }
            >
              Watermelon
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hitSlop={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#fff",
                    "height": 50,
                    "justifyContent": "center",
                    "marginTop": 0.5,
                  },
                  Object {},
                ],
              ]
            }
            testID={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "red",
                }
              }
            >
              Durian
            </Text>
          </View>
        </ScrollView>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "height": 50,
                  "justifyContent": "center",
                  "marginTop": 0.5,
                },
                Object {
                  "marginTop": 6,
                },
              ],
            ]
          }
          testID={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontSize": 18,
                },
                Object {
                  "color": "#007aff",
                  "fontWeight": "700",
                },
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
